CT-1728 Remove sensitivity notice for SAR cases

stephenrichards opened the pull request:
This is acheived by setting the escalation date to the day before
the creation date.  This prevents the sensitivity banner from ever
being displayed, as the escalation date is always in the past.

misaka commented on app/policies/case_policy.rb:
```diff
@@ -8,15 +8,19 @@ def initialize(user, kase)
+    self.case.requires_clearance? && responder_attachable?
+  end
+
+  def can_upload_response_and_approve?
```
Can we make this name clearer? `can_upload_cleared_response_and_approve` for example?

stephenrichards replied:
Apart from the cleared/approve (we need to choose one and stick to it - I would favour approve because I think that gives  more of a flavour of what it does, rather than clear which has other meanings), I think the name is correct: it's asking does the user have permission to upload a response and clear.

If you feel strongly about it, I will, but I do think it conveys the sense.  Ask Al, see what  he thinks.

misaka replied:
Ah, sorry, I read it wrong. I agree with the use of `approved` over `cleared`, but I thought the permission being requested is to be able to upload a response and to be able to approve a response, not to upload an approved response. I think `can_upload_approved_response` is clearer, but I guess `can_upload_response_and_approve` is clear enough if I think about it.

misaka commented on spec/lib/case_state_machine_spec.rb:
```diff
@@ -333,6 +341,39 @@ def event(event_name)
   end
 
+
+  context 'approvals' do
```
I think that if we need to create our own little microcosm-context here instead of relying on the global `let` statements at the top of the file, this is probably a sign that we should just move these tests into a separate file ... `case_state_machine/approvals_spec.rb` or something like that.

misaka commented on spec/views/cases/_case_status_html_slim_spec.rb:
```diff
@@ -5,8 +5,8 @@
                    status: "Needs reassigning",
-                   internal_deadline: DateTime.now.strftime("%e %b %Y"),
-                   external_deadline: (DateTime.now + 10.days).strftime("%e %b %Y"),
+                   internal_deadline: DateTime.now.strftime("%-d %b %Y"),
```
This should re-use `Settings.default_date_format` as above, if it can of course.

stephenrichards committed:
CT-1728 Remove sensitivity notice for SAR cases

This is acheived by setting the escalation date to the day before
the creation date.  This prevents the sensitivity banner from ever
being displayed, as the escalation date is always in the past.
